Flaky test: TestSubmitSegment_Success

This test failed once but passed after repeating it multiple times.
```
--- FAIL: TestSubmitSegment_Success (0.02s)
    segment_rpc_test.go:1334: 
        	Error Trace:	segment_rpc_test.go:1334
        	Error:      	"7.667e-06" is not less than "5.7034e-06"
        	Test:       	TestSubmitSegment_Success
```
This can cause CI to become unpredictable.  It isn't an urgent problem right now - just creating an issue so that it can be tracked.

--- a/server/segment_rpc.go
+++ b/server/segment_rpc.go
@@ -464,7 +464,7 @@ func SubmitSegment(sess *BroadcastSession, seg *stream.HLSSegment, nonce uint64)
 	return &ReceivedTranscodeResult{
 		TranscodeData: tdata,
 		Info:          tr.Info,
-		LatencyScore:  transcodeDur.Seconds() / seg.Duration,
+		LatencyScore:  tookAllDur.Seconds() / seg.Duration,
 	}, nil
 }
 

--- a/server/segment_rpc_test.go
+++ b/server/segment_rpc_test.go
@@ -13,6 +13,7 @@ import (
 	"net/url"
 	"strings"
 	"testing"
+	"time"
 
 	ethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/golang/protobuf/proto"
@@ -1276,7 +1277,7 @@ func TestSubmitSegment_Success(t *testing.T) {
 		if runChecks != nil {
 			runChecks(r)
 		}
-
+		time.Sleep(20 * time.Millisecond)
 		w.WriteHeader(http.StatusOK)
 		w.Write(buf)
 	})
@@ -1295,16 +1296,18 @@ func TestSubmitSegment_Success(t *testing.T) {
 
 	assert := assert.New(t)
 
+	segData := pm.RandBytes(1024)
+
 	runChecks = func(r *http.Request) {
 		assert.Equal("video/MP2T", r.Header.Get("Content-Type"))
 
 		data, err := ioutil.ReadAll(r.Body)
 		require.Nil(err)
 
-		assert.Equal([]byte("dummy"), data)
+		assert.Equal(segData, data)
 	}
 
-	noNameSeg := &stream.HLSSegment{Data: []byte("dummy")}
+	noNameSeg := &stream.HLSSegment{Data: segData}
 	tdata, err := SubmitSegment(s, noNameSeg, 0)
 
 	assert.Nil(err)
@@ -1314,7 +1317,7 @@ func TestSubmitSegment_Success(t *testing.T) {
 	assert.Nil(tdata.Info)
 
 	// Check that latency score calculation is different for different segment durations
-	// The transcode duration calculated in SubmitSegment should be about the same across all calls
+	// The round trip duration calculated in SubmitSegment should be about the same across all calls
 	noNameSeg.Duration = 5.0
 	tdata, err = SubmitSegment(s, noNameSeg, 0)
 	assert.Nil(err)

